src/treelearner/cuda/cuda_single_gpu_tree_learner.hpp

```diff
@@ -0,0 +1,143 @@
+/*!
+ * Copyright (c) 2021 Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE file in the project root for
+ * license information.
+ */
+#ifndef LIGHTGBM_NEW_CUDA_TREE_LEARNER_HPP_
+#define LIGHTGBM_NEW_CUDA_TREE_LEARNER_HPP_
+
+#include <memory>
+#include <vector>
+
+#ifdef USE_CUDA_EXP
+
+#include "cuda_leaf_splits.hpp"
+#include "cuda_histogram_constructor.hpp"
+#include "cuda_data_partition.hpp"
+#include "cuda_best_split_finder.hpp"
+
+#include "../serial_tree_learner.h"
```

Comment:
full path include.

Reply:
I think we don't have a better way to include this. Since `serial_tree_learner.h` is not in the include path according to CMakeLists.txt. But I've fixed all other relative path include introduced by this PR.